cli/src/commands/describe.rs

```diff
@@ -67,35 +77,96 @@ pub(crate) fn cmd_describe(
     args: &DescribeArgs,
 ) -> Result<(), CommandError> {
     let mut workspace_command = command.workspace_helper(ui)?;
-    let commit = workspace_command.resolve_single_rev(&args.revision)?;
-    workspace_command.check_rewritable([commit.id()])?;
-    let description = if args.stdin {
+    let commits = workspace_command
+        .resolve_some_revsets_default_single(&args.revisions)?
+        .into_iter()
+        .collect_vec();
+    if commits.is_empty() {
+        return Err(user_error("Empty revision set"));
+    }
+    workspace_command.check_rewritable(commits.iter().ids())?;
+    let shared_description = if args.stdin {
         let mut buffer = String::new();
         io::stdin().read_to_string(&mut buffer).unwrap();
-        buffer
+        Some(buffer)
     } else if !args.message_paragraphs.is_empty() {
-        join_message_paragraphs(&args.message_paragraphs)
-    } else if args.no_edit {
-        commit.description().to_owned()
+        Some(join_message_paragraphs(&args.message_paragraphs))
     } else {
-        let template =
-            description_template_for_describe(ui, command.settings(), &workspace_command, &commit)?;
-        edit_description(workspace_command.repo(), &template, command.settings())?
+        None
     };
-    if description == *commit.description() && !args.reset_author {
-        writeln!(ui.status(), "Nothing changed.")?;
+
+    // TODO: Allow editing description of multiple commits in a single editor?
```

Comment:
Fwiw, I would really like that feature for a few reasons: 

* Some editors take several seconds to start
* Allows find&replace and copy&paste across all descriptions
* Easier to abort

Not asking you to do anything in this PR except for maybe dropping the question mark in the TODO :)

Reply:
Perhaps, there will be three modes:
* edit multiple commits within a single file
* edit multiple commits one by one
* edit multiple commits by multiple files, but edit them all at once

I also like the first option.

Reply:
The one-by-one mode seems better handled by the shell (should be something like `for i in $(jj log -T '{commit_id}\n' -r SET); do jj describe -r "$i" || break; done`?) so I would suggest ultimately not putting it in core jj unless it turns out to be used by many people. Maintaining all the different modes might be a burden. (Ideally I guess jj would also offer some facility to run an arbitrary command in a transaction and abort the transaction if the command fails.)

Reply:
Makes sense, I will update this to just edit multiple commits within a single file.

Reply:
imo editing them in the same file is what actually makes this useful
so you can e.g. use search replace or vim macros (or multiple cursors in other editors)
if you would open several instances of the editor that could also be hacked together with a shell command ig

Reply:
I've updated the PR to edit multiple descriptions within a single file. I'm using `JJ: describe [commit_id_prefix]` as the separator between commit descriptions.